src/vmm/src/guest_config/x86_64/mod.rs

```diff
@@ -106,10 +109,17 @@ impl CpuConfiguration {
             mut msrs,
         } = self;
 
-        let guest_cpuid = match &mut cpuid {
-            Cpuid::Intel(cpuid) => &mut cpuid.0,
-            Cpuid::Amd(cpuid) => &mut cpuid.0,
-        };
+        // Check whether the vendor specified in CPU template matches the actual vendor.
```

Comment:
Sure, I can. This part does two things: 1. check the CPU vendor and 2. retrieve internal BTreeMap. So it might be better to implement it as a method of `Cpuid` type (like `inner_mut_with_vendor_check()` , because we have already had `inner()` and `inner_mut()` methods there).